packages/react-reconciler/src/ReactFiberHydrationContext.old.js

```diff
@@ -556,6 +572,16 @@ function popHydrationState(fiber: Fiber): boolean {
   return true;
 }
 
+function hasUnhydratedTailNodes() {
+  return isHydrating && nextHydratableInstance !== null;
+}
+
+function warnUnhydratedNextInstance(fiber: Fiber) {
```

Comment:
The "NextInstance" part is an implementation detail of this file. Maybe go with the other terminology you had. "warnUnhydratedTailNodes"

Reply:
Sure, one thing though I only warn for the nextHydratableInstance instead of warning for them in a loop because I know the implementation uses a flag to only warn the first time anyways. Though with the error logging API maybe we will want it to there every time so maybe I should add the loop here to warn for all for all of the tail nodes and not just the next one.

Similarly above we throw after the first one is warned for so we don't go through through the warning function for all of the tail nodes there either.

Reply:
The reason it logged them in a loop is so that the implementation in the host config could batch them up and build a tree that shows all the errors in one message. We never landed the PR that did that but we should.

So this should probably call it in a loop too.